src/ai_migrate/migrate.py

```diff
@@ -539,6 +499,25 @@ async def _run(
     if not examples:
         raise FileNotFoundError("No valid example pairs found in examples directory")
 
+    from .example_selector import select_relevant_examples
+
+    log("[agent] Running example selection analysis...")
+    selection_result = await select_relevant_examples(target_files, examples, client)
```

Comment:
should we put it behind a flag?

Reply:
hm good point. which would be the default tho?